src/api/services/participantsService.ts

```diff
@@ -127,7 +127,16 @@ const updateParticipantAssociatedRequestTypes = async (
   await participant.$relatedQuery('types', trx).relate(participantApprovalPartial.types);
 };
 
-export const updateParticipantAndTypes = async (
+const updateParticipantAssociatedRequestApiRoles = async (
+  participant: Participant,
+  participantApprovalPartial: z.infer<typeof ParticipantApprovalPartial>,
+  trx: TransactionOrKnex
+) => {
+  await participant.$relatedQuery('apiRoles', trx).unrelate();
```

Comment:
I'm still a little cautious of Objection.js and knex - have you checked the SQL this generates?

Reply:
![image](https://github.com/IABTechLab/uid2-self-serve-portal/assets/152250444/8c787f01-dd05-4be0-b0b0-541648e1e48f)
It removes any link between the two tables for that objects.
["Doesn't delete the items. Only removes the connection. For ManyToMany relations this deletes the join row from the join table. For other relation types this sets the join columns to null."](https://vincit.github.io/objection.js/api/query-builder/mutate-methods.html#unrelate)